hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java

```diff
@@ -234,48 +234,50 @@ public void promoteSuspended() {
+     * @param mergingEntry the {@link MergingEntryHolder} instance to merge
      * @param mergePolicy  the {@link SplitBrainMergePolicy} instance to apply
      * @return the used {@link ScheduledTaskDescriptor} if merge is applied, otherwise {@code null}
      */
-    public ScheduledTaskDescriptor merge(MergingValueHolder<ScheduledTaskDescriptor> mergingValue,
+    public ScheduledTaskDescriptor merge(MergingEntryHolder<String, ScheduledTaskDescriptor> mergingEntry,
                                          SplitBrainMergePolicy mergePolicy) {
         SerializationService serializationService = nodeEngine.getSerializationService();
-        serializationService.getManagedContext().initialize(mergingValue);
+        serializationService.getManagedContext().initialize(mergingEntry);
         serializationService.getManagedContext().initialize(mergePolicy);
 
-        // try to find an existing item with the same value
-        ScheduledTaskDescriptor match = null;
-        for (ScheduledTaskDescriptor item : tasks.values()) {
-            if (mergingValue.getValue().equals(item)) {
-                match = item;
+        // try to find an existing task with the same definition
+        ScheduledTaskDescriptor existingTask = null;
+        for (ScheduledTaskDescriptor task : tasks.values()) {
+            if (mergingEntry.getValue().equals(task)) {
+                existingTask = task;
                 break;
             }
         }
-
-        ScheduledTaskDescriptor merged;
-        if (match == null) {
-            // Missing incoming entry
-            merged = mergePolicy.merge(mergingValue, null);
-            if (merged != null) {
-                enqueueSuspended(merged, false);
+        if (existingTask == null) {
+            ScheduledTaskDescriptor newTask = mergePolicy.merge(mergingEntry, null);
+            if (newTask != null) {
+                enqueueSuspended(newTask, false);
+                return newTask;
             }
         } else {
-            // Found a match -> real merge
-            MergingValueHolder<ScheduledTaskDescriptor> existingValue = createMergeHolder(serializationService, match);
-            merged = mergePolicy.merge(mergingValue, existingValue);
-            if (merged != null && !merged.equals(match)) {
-                // Cancel matched one, before replacing it
-                match.cancel(true);
-                enqueueSuspended(merged, true);
-            } else {
-                merged = null;
+            MergingEntryHolder<String, ScheduledTaskDescriptor> existingEntry
+                    = createMergeHolder(serializationService, existingTask);
+            ScheduledTaskDescriptor newTask = mergePolicy.merge(mergingEntry, existingEntry);
+            if (newTask != null) {
+                boolean hasSameDescriptor = newTask.equals(existingTask);
+                boolean hasSameResult = existingTask.getTaskResult().equals(newTask.getTaskResult());
```

Comment:
This is the fix for the ignored different `TaskResult`.

Reply:
Are you sure this is working? `ScheduledTaskResult` doesn't have `equals` implemented. So you are basically comparing refs, which will always fail. Also I don't see the value on comparing the result, because you don't know at what stage its of the tasks is at. Started/failed/subsequent-execution etc. Depending on the policy, if a task has the same descriptor that should be enough to make a decision. Am I missing sth?

Reply:
Hmm, but I'm starting tasks with the same descriptor but different results in the PassThrough scenario. In this case the tasks from the smaller cluster should win, but since the descriptor is the same, nothing is overwritten. I was able to prove this with the different results (otherwise they look of course the same).

Without that fix the PassThrough test failed. Either my expectation is wrong or the actual merging logic.

Reply:
If I just check `if (!hasSameDescriptor)` then we get the assert from the PR description for `PassThroughMergePolicy`:
```
java.lang.AssertionError: 
Expected :42.0
Actual   :100.0
```
That's the result from the larger cluster, which is not what `PassThroughMergePolicy` does (it returns the descriptor of the merging entry.

Reply:
Yes I see the issue, I believe the right fix for this is to check for `==` rather than `equals` on the descriptor, since so far we only needed the task-def to be the same for a descriptor to be the same. However, for this use-case, the descriptor will be the same, but if it got changed by `PassThrough` then the `==` should fail. Have a go with that, and I ll see if i can think of anything wrong with it.

Reply:
`if (newTask != null && newTask != existingTask)` seems to work. I changed the PR and added a comment. But we really have to think about this, since all other merge methods work with `equals()`.

Reply:
The semantics of the `descriptor` are different on purpose. We want to know when two tasks are the same by just checking their definition, but, we on-purpose ignore the result, mainly because tasks might be schedule `atFixedRate` so the result plays no role there, since we can't expect two descriptors to be in total sync. So as long as the definition is the same we assume same task. I believe the solution we applied here is correct, Alternatively, we could rename current `equals` to `sameDefinition` and then introduce another `equals` which relies only on the ref check that you did here. That's probably a change for 3.11 to have more time to properly test it.
Anyhow, good finding :+1:

Reply:
The executors are not just a normal data structures. Your explanation makes totally sense. I think for now this fix should work. I also see no problems with a later fix, since (at least for merging) it's a local thing, so the merge operation doesn't need to be changed.